packages/core/strapi/lib/core-api/controller/index.d.ts

```diff
@@ -23,5 +23,5 @@ export interface CollectionTypeController extends Controller {
 }
 
 export type GenericController = Partial<Controller> & {
-  [method: string | number | symbol]: (ctx: Context) => unknown;
+  [method: string | number | symbol]: (ctx: Context, next: Next) => unknown;
```

Comment:
@Convly if we don't provide this parameter in here, it would break if we put:
```
factories.createCoreController('api::home.home', () => ({
  find(ctx, next) {
    return {};
  },
}));
```

Maybe instead of force it we can put it as optional 
```suggestion
  [method: string | number | symbol]: (ctx: Context, next?: Next) => unknown;
```

Reply:
`next` isn't actually a required parameter from what I can see, in fact I'm having trouble finding a controller that even uses it rather than just returning a value. Is there somewhere that it's actually required?

Reply:
As it is a parameter, it's not going to be required, you can ignore it and pass just the context if you want, so I don't think it would matter in this case to put it as optional or not